pkg/controller/gitopsconfig/controller_test.go

```diff
@@ -562,3 +564,66 @@ func findDeleteJob(cl client.Client) batch.Job {
 	}
 	return batch.Job{}
 }
+
+func TestCreateJob(t *testing.T) {
+	// This flag is needed to let the reconciler know that the CRD has been initialized
+	gitops.Annotations = map[string]string{"gitopsconfig.eunomia.kohls.io/initialized": "true"}
+	// Set trigger type to Change
+	gitops.Spec.Triggers = []gitopsv1alpha1.GitOpsTrigger{
+		{
+			Type: "Change",
+		},
+	}
+	// Objects to track in the fake client.
+	objs := []runtime.Object{
+		gitops,
+	}
+	// Register operator types with the runtime scheme.
+	s := scheme.Scheme
+	s.AddKnownTypes(gitopsv1alpha1.SchemeGroupVersion, gitops)
+	// Initialize fake client
+	cl := fake.NewFakeClient(objs...)
+	r := &Reconciler{client: cl, scheme: s}
+
+	nsn := types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}
+
+	req := reconcile.Request{
+		NamespacedName: nsn,
+	}
+
+	numberOfJobs := 2
+	var wg sync.WaitGroup
+	for i := 1; i <= numberOfJobs; i++ {
+		wg.Add(1)
+		// in order to replicate reconciler being called for creating an another job.
+		go func() {
+			r.Reconcile(req)
```

Comment:
Instead of running Reconcile, could we just call `r.createJob`? or are there reasons why we cannot or should not do this?

Reply:
I followed this pattern as we have used reconcile in all other test cases as a standard rather than using a createJob.